### PR TITLE
Clean up server-side translations, remove global variable

### PR DESCRIPTION
**Pull request by rubenwardy**
## To do

- [ ] Testing

### CODE REVIEW

**Comment by rubenwardy** [comment]
Alternative:

```c++
std::ifstream is(real_path, std::ios::binary | std::ios::ate);
if (!is.good()) {
	return false;
}

auto size = is.tellg();
std::string contents(size, '\0');
is.seekg(0);
is.read(&contents[0], size);
```

File: src/filesys.cpp [hunk 1]
```diff
@@ -750,6 +750,21 @@ bool safeWriteToFile(const std::string &path, const std::string &content)
...
 
+bool ReadFile(const std::string &path, std::string &out)
+{
+	std::ifstream is(path, std::ios::binary | std::ios::ate);
+	if (!is.good()) {
+		return false;
+	}
+
+	auto size = is.tellg();
+	out = std::string(size, '\0');
```
Comment (sfan5) [inline src/filesys.cpp:761]: should be `out.resize(size);`

File: src/server.cpp [hunk 2]
```diff
@@ -2433,31 +2433,14 @@ bool Server::addMediaFile(const std::string &filename,
...
-		if (!fis.good()) {
-			bad = true;
-			break;
-		}
-	}
-	if (bad) {
+	if (!fs::ReadFile(filepath, filedata)) {
 		errorstream << "Server::addMediaFile(): Failed to read \""
-				<< filename << "\"" << std::endl;
+					<< filename << "\" for reading" << std::endl;
```
Comment (sfan5) [inline src/server.cpp:2439]: >Failed to read "..." for reading

**Comment by rubenwardy** [comment]
Updated

**Review by sfan5 — APPROVED** [review]
Looks good.

**Pull request merged**